plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/IcebergTestUtils.java

```diff
@@ -101,40 +108,45 @@ private static boolean checkOrcFileSorting(Supplier<OrcDataSource> dataSourceSup
         OrcReaderOptions readerOptions = new OrcReaderOptions();
         try (OrcDataSource dataSource = dataSourceSupplier.get()) {
             OrcReader orcReader = OrcReader.createOrcReader(dataSource, readerOptions).orElseThrow();
-            String previousMax = null;
-            OrcColumnId sortColumnId = orcReader.getRootColumn().getNestedColumns().stream()
+            OrcColumn sortColumn = orcReader.getRootColumn().getNestedColumns().stream()
                     .filter(column -> column.getColumnName().equals(sortColumnName))
-                    .collect(onlyElement())
-                    .getColumnId();
-            List<StripeStatistics> statistics = orcReader.getMetadata().getStripeStatsList().stream()
-                    .map(Optional::orElseThrow)
-                    .collect(toImmutableList());
-            verify(statistics.size() > 1, "Test must produce at least two row groups");
-
-            for (StripeStatistics stripeStatistics : statistics) {
-                // TODO: This only works if the sort column is a String
-                StringStatistics columnStatistics = stripeStatistics.getColumnStatistics().get(sortColumnId).getStringStatistics();
-
-                Slice minValue = columnStatistics.getMin();
-                Slice maxValue = columnStatistics.getMax();
-                if (minValue == null || maxValue == null) {
-                    throw new IllegalStateException("ORC files must produce min/max stripe statistics");
-                }
-
-                if (previousMax != null && previousMax.compareTo(minValue.toStringUtf8()) > 0) {
-                    return false;
+                    .collect(onlyElement());
+            Type sortColumnType = getSpiType(sortColumn.getColumnType().getOrcTypeKind());
+            try (OrcRecordReader recordReader = orcReader.createRecordReader(
+                    List.of(sortColumn),
+                    List.of(sortColumnType),
+                    OrcPredicate.TRUE,
+                    UTC,
+                    newSimpleAggregatedMemoryContext(),
+                    INITIAL_BATCH_SIZE,
+                    RuntimeException::new)) {
+                String previousMax = "";
+                for (Page page = recordReader.nextPage(); page != null; page = recordReader.nextPage()) {
+                    Block block = page.getLoadedPage().getBlock(0);
+                    for (int position = 0; position < block.getPositionCount(); position++) {
+                        String current = (String) sortColumnType.getObjectValue(TestingConnectorSession.SESSION, block, position);
```

Comment:
`String current` -> `Object current`
Initialize it to be `null` and handle that in the `if` below